docs/platforms/react-native/migration/before-v5.mdx

```diff
@@ -0,0 +1,302 @@
+
+## From 3.0.x to 3.1.x
+
+`ReactNavigationV5Instrumentation` was renamed to `ReactNavigationInstrumentation` and supports every version of React Navigation from v5 onwards, including v6. You only need to change the name wherever you call the constructor for the instrumentation:
+
+```javascript
+// Old
+const routingInstrumentation = new Sentry.ReactNavigationV5Instrumentation();
+// New
+const routingInstrumentation = new Sentry.ReactNavigationInstrumentation();
+```
+
+## From 2.x to 3.x
+
+There are no changes needed when migrating from versions `2.x` to `3.x`, although you will need to make sure that you run `pod install` on iOS and to rebuild your app on both platforms.
+
+### Event origin and environment tags
+
+When upgrading from prior versions to `3.x`, you may see a mismatch in the values of the `event.environment` tag compared to events sent from the SDK prior to this version. Events that originate from native iOS code will now have the `event.environment` tag set to `native`. Read more about these tags on the [event information guide](/platforms/react-native/usage/event-information/#event-origin-and-environment-tags).
+
+## From 2.4.x to 2.5.x
+
+The breaking changes are relevant only to Android. There are no breaking changes for other platforms.
+
+### Android Specific Changes
+
+Sentry React Native version `2.5.0` depends on Sentry Android `5.0.0`. Please [refer to the Android migration guide for Android-specific changes](/platforms/android/migration/#migrating-from-iosentrysentry-android-430-to-iosentrysentry-android-500).
+
+`Settings.Secure.ANDROID_ID` has been removed and replaced with a randomly-generated `installationId`. This may affect the number of unique users displayed on the the Issues page and Alerts. If you always set a custom user using `Sentry.setUser(customUser)`, the behavior has not changed. While you don't have to make any update, if you want to maintain the old behavior, use the following code snippet which makes use of the [react-native-device-info](https://github.com/react-native-device-info/react-native-device-info) library.
+
+```javascript
+import { Platform } from "react-native";
+import DeviceInfo from "react-native-device-info";
+
+import * as Sentry from "@sentry/react-native";
+
+Sentry.init({
+  // ...
+});
+
+// Only add the event processor on Android
+if (Platform.OS === "android") {
+  Sentry.addGlobalEventProcessor((event) => {
+    // Get the ANDROID_ID
+    const id = DeviceInfo.getUniqueId();
+
+    // If the user does not exist, set the id to be the unique id.
+    if (!event.user) {
+      event.user = { id };
+    }
+
+    return event;
+  });
+}
+```
+
+## React Navigation Instrumentation from &lt;2.3.0
+
+We changed the initialization method for the React Navigation v5 and above routing instrumentation to avoid a potential issue when using linking. You now register the navigation container inside the `onReady` prop passed to the `NavigationContainer`.
```

Comment:
```suggestion
We changed the initialization method for the React Navigation v5 and above, routing instrumentation to avoid a potential issue when using linking. You now register the navigation container inside the `onReady` prop passed to the `NavigationContainer`.
```